src/useCases/users.js

```diff
@@ -0,0 +1,67 @@
+const User = require('../models/users')
+const bcrypt = require('../lib/bcrypt')
+const jwt = require('../lib/jwt')
+
+function getAll () {
+    return User.find({})
+}
+
+
+function create (UserData) {
+    return User.create(UserData)
+  }
+
+  function deleteById (id) {
+    return User.findByIdAndRemove(id)
+  }
+  
+  function updateById (id, newUserData) {
+    return User.findByIdAndUpdate(id, newUserData, { new: true })
+  }
+
+  function getById (id) {
+    return User.findById(id)
+  }
+
+async function signUp (dataUser) {
+    // name, email, password
+    // valudar que no exista un usuario con el email
+    const {email, password} = dataUser
+
+    const userFound = await User.findOne({email}) //objeto {name, password, email}
+    // si no lo encuentra regresa un undefined
+
+    if(userFound) throw new Error('User already exists')
+
+    // encriptar mi constraseña
+    const passwordEncrypted = await bcrypt.hash(password)
+
+    return User.create({ email, password: passwordEncrypted})
```

Comment:
@AldaMayorgaDev Esta bien esta parte, solo que si mandan el nombre rfc u otro data no lo guardará en la BD
por lo que puede quedar así.

```suggestion
    return User.create({...dataUser, password: passwordEncrypted})
```

Es decir propagar lo que venga en el `dataUser` y sobre escribir el password con el password encriptado